web/src/user/UserInterface.ts

```diff
@@ -67,118 +124,62 @@ export class UserInterface extends Interface {
-                }
-                .display-none {
-                    display: none;
-                }
-                .pf-c-brand {
-                    min-height: 32px;
-                    height: 32px;
-                }
-                .has-notifications {
-                    color: #2b9af3;
-                }
-                .background-wrapper {
-                    height: 100vh;
-                    width: 100%;
-                    position: fixed;
-                    z-index: -1;
-                    top: 0;
-                    left: 0;
-                    background-color: var(--pf-c-page--BackgroundColor) !important;
-                }
-                .background-default-slant {
-                    background-color: white; /*var(--ak-accent);*/
-                    clip-path: polygon(0 0, 100% 0, 100% 100%, 0 calc(100% - 5vw));
-                    height: 50vh;
-                }
-                :host([theme="dark"]) .background-default-slant {
-                    background-color: black;
-                }
-                ak-locale-context {
-                    display: flex;
-                    flex-direction: column;
-                }
-            `,
+            customStyles,
         ];
     }
 
-    constructor() {
-        super();
-        this.ws = new WebsocketClient();
-        window.addEventListener(EVENT_NOTIFICATION_DRAWER_TOGGLE, () => {
-            this.notificationDrawerOpen = !this.notificationDrawerOpen;
-            updateURLParams({
-                notificationDrawerOpen: this.notificationDrawerOpen,
-            });
-        });
-        window.addEventListener(EVENT_API_DRAWER_TOGGLE, () => {
-            this.apiDrawerOpen = !this.apiDrawerOpen;
-            updateURLParams({
-                apiDrawerOpen: this.apiDrawerOpen,
-            });
-        });
-        window.addEventListener(EVENT_WS_MESSAGE, () => {
-            this.firstUpdated();
-        });
-        configureSentry(true);
-    }
+    @property({ type: Object })
+    uiConfig!: UIConfig;
 
-    async firstUpdated(): Promise<void> {
-        this.me = await me();
-        const notifications = await new EventsApi(DEFAULT_CONFIG).eventsNotificationsList({
-            seen: false,
-            ordering: "-created",
-            pageSize: 1,
-            user: this.me.user.pk,
-        });
-        this.notificationsCount = notifications.pagination.count;
-    }
+    @property({ type: Object })
+    me!: SessionUser;
 
-    render(): TemplateResult {
-        if (!this.uiConfig || !this.me) {
-            return html``;
-        }
-        let userDisplay = "";
+    @property({ type: Boolean, reflect: true })
+    notificationDrawerOpen = false;
+
+    @property({ type: Boolean, reflect: true })
+    apiDrawerOpen = false;
+
+    @property({ type: Number })
+    notificationsCount = 0;
+
+    @property({ type: Object })
+    brand!: CurrentBrand;
+
+    get userDisplayName() {
```

Comment:
Turns this into a switch _expression_, so Typescript can type it and assert conditional exhaustion correctly!